x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -104,6 +107,34 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
     return location;
   };
 
+  const getAnomalySourceMapsLink = async (
+    anomaly: AnomaliesTableRecord,
+    sourceIndicesWithGeoFields: {
+      [key: string]: { [key: string]: { geoFields: string[]; dataViewId: string } };
+    }
+  ) => {
+    // Create a layer for each of the geoFields
+    const initialLayers = getInitialSourceIndexFieldLayer(
+      sourceIndicesWithGeoFields[anomaly.jobId]
+    );
+
+    const locator = share.url.locators.get(MAPS_APP_LOCATOR);
+    const location = await locator?.getLocation({
+      initialLayers,
+      timeRange: data.query.timefilter.timefilter.getTime(),
+      filters: data.query.filterManager.getFilters(),
```

Comment:
If there are additional influencer fields, they should also be passed to the map. There could be multiple influencer field values, so ideally all the values should be passed - such as in a 'one of' filter

<img width="320" alt="image" src="https://user-images.githubusercontent.com/7405507/186405589-f195fe62-f05d-4916-8836-80e861f359c0.png">


or if that's not possible in the query as e.g.

```
http.response.status_code:200 and (source.address:149.202.247.173 or source.address:197.253.6.253)
```

Reply:
Added to query in 3c7de433427453bfe4d8dd136fdafea492611881